portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,30 +35,41 @@
   @Override
   public void init() {
     messages = new ArrayList<>();
-    messages.add("Once upon a time...");
-    messages.add("Twice upon two times...");
-    messages.add("Thrice upon three times...");
   }
 
+  /** Gets the messages stored in the messages variable in JSON format */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
     response.getWriter().println(convertToJson(messages));
   }
 
+  /** Posts a message retrieved from the form input adding it to the messages variable */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Get the message from the form and add it to the array */
+    String message = request.getParameter("text-input");
+    messages.add(message);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
   /**
    * Converts an array into a JSON string using manual String concatentation.
+   * Keys will be a String called 'string' plus the number of the String.
+   * Values will be a String containing a message (user comment).
    */
   private String convertToJson(ArrayList messages) {
     String json = "{";
-    json += "\"string1\": ";
-    json += "\"" + messages.get(0) + "\"";
-    json += ", ";
-    json += "\"string2\": ";
-    json += "\"" + messages.get(1) + "\"";
-    json += ", ";
-    json += "\"string3\": ";
-    json += "\"" + messages.get(2) + "\"";
+    int i;
```

Comment:
Quoting your last PR:

'The reason I declared it outside of the for loop is because I use it again after the for loop to access the last element. I could also write 'messages.size() - 1' instead of 'i', and therefore, I would not need to use it outside the scope of the for loop. Should I make that change?'

Yeah that sounds like a better way to do it! In fact, I'd suggest within the for loop just checking if i == messages.size() - 1, and only adding the comma if this is false, so you don't have to have the extra bit of logic outside the loop (and therefore extending the for loop to 'i < messages.size()'.

Reply:
Ok, sounds good! Done